### PR TITLE
Auto updater

### PR DESCRIPTION
**Pull request by Njanderson**
image

First contribution to any open source project. Hoping that this solves the issue, which is issue #7339 about auto updating,

### CODE REVIEW

File: src/browser/auto-update-manager.coffee [hunk 1]
```diff
@@ -1,6 +1,7 @@
 autoUpdater = null
 _ = require 'underscore-plus'
 {EventEmitter} = require 'events'
+atom = require 'atom'
```
Comment (lee-dohm) [inline src/browser/auto-update-manager.coffee]: I'm not sure if `atom` can be required here. The files in the `browser` directory execute in the main process, not the renderer process where most of the rest of Atom runs, including all the places where `require 'atom'` is usually used.

**Comment by Njanderson** [comment]
Thanks for the feedback; I removed that requires statement. 

Love to work on anything you suggest I do to improve this PR and contribute to Atom in general.

**Comment by lee-dohm** [comment]
The problem is that if you remove the `require 'atom'`, then you shouldn't be able to call `atom.config.get()` later in the same file. This is why making changes in the main process is a little sticky :smile: I'm not sure I have a good solution, which is why I called in the feedback team.

You can use [this search](https://github.com/search?utf8=%E2%9C%93&q=user%3Aatom+label%3Abeginner+is%3Aissue+is%3Aopen&type=Issues&ref=searchresults) to find anything that the triage team has labeled as "beginner". Items carrying that label are things that we think are a great starting point for people wanting to get their feet wet :sunglasses:

**Comment by Njanderson** [comment]
Awesome! Thank you so much Lee! I'll take a look right now.

**Comment by batjko** [comment]
For my benefit, is the `@setupAutoUpdater()` call not just checking for updates (as opposed to actually auto-updating)?

Because if that's the case, the option in the Settings View might be better named as "Auto Update Check", which sounds terrible, but you know what I'm getting at.

**Pull request closed without merging**